faunadb/client.go

```diff
@@ -130,8 +134,19 @@ func NewFaunaClient(secret string, configs ...ClientConfig) *FaunaClient {
 	}
 
 	if client.http == nil {
+		dial := func(network, addr string, cfg *tls.Config) (net.Conn, error) {
+			return net.Dial(network, addr)
+		}
+		if len(client.endpoint) > 5 && client.endpoint[:5] == "https" {
+			dial = nil
+		}
+		transport := &http2.Transport{
+			DialTLS:   dial,
+			AllowHTTP: true,
+		}
+
 		client.http = &http.Client{
-			Timeout: requestTimeout,
```

Comment:
Does this mean we're no longer enforcing requests timeout?

Reply:
I was thinking about streaming when removed this timeout. I will return timeout for all requests by default.